src/main/java/com/percero/agents/sync/dao/IDataAccessObject.java

```diff
@@ -0,0 +1,41 @@
+package com.percero.agents.sync.dao;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+
+import com.percero.agents.sync.exceptions.SyncException;
+import com.percero.agents.sync.metadata.MappedField;
+import com.percero.agents.sync.vo.ClassIDPair;
+import com.percero.agents.sync.vo.ClassIDPairs;
+import com.percero.framework.vo.IPerceroObject;
+import com.percero.framework.vo.PerceroList;
+
+public interface IDataAccessObject<T extends IPerceroObject> {
```

Comment:
Yep, pretty true for everything.  But definitely this interface since it would be a very common interface for devs.